### PR TITLE
Minor version upgrade for Spark 3.4. (Updating to Spark 3.4.1)

### PR DESCRIPTION
**Pull request by mahendruajay**
*Issue #, if available:*

*Description of changes:*

Build Spark 3.4.1. Changes:
1. Consume EMR 6.15.0
2. Minor update to make clean
3. Update kernel and other libs to address CVEs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,8 +31,10 @@ all: build test
 # Downloads EMR packages. Skips if the tar file containing EMR packages has been made.
 
 init:
+	python --version
+	pip install --upgrade pip
 	# pipenv > 2022.4.8 fails to build smspark
-	python3 -m pip install pipenv==2022.4.8
+	python -m pip install pipenv==2022.4.8
```
Comment (shenxiaoxu) [inline Makefile:37]: why do we remove python 3 from the command?

File: spark/processing/3.4/py3/hadoop-config/spark-defaults.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+spark.driver.extraClassPath      /usr/lib/hadoop-lzo/lib/*:/usr/lib/hadoop/hadoop-aws.jar:/usr/share/aws/aws-java-sdk/*:/usr/share/aws/aws-java-sdk-v2/*:/usr/share/aws/emr/emrfs/conf:/usr/share/aws/emr/emrfs/lib/*:/usr/share/aws/emr/emrfs/auxlib/*:/usr/share/aws/emr/goodies/lib/emr-spark-goodies.jar:/usr/share/aws/emr/security/conf:/usr/share/aws/emr/security/lib/*:/usr/share/aws/hmclient/lib/aws-glue-datacatalog-spark-client.jar:/usr/share/java/Hive-JSON-Serde/hive-openx-serde.jar:/usr/share/aws/sagemaker-spark-sdk/lib/sagemaker-spark-sdk.jar:/usr/share/aws/emr/s3select/lib/emr-s3-select-spark-connector.jar
```
Comment (shenxiaoxu) [inline spark/processing/3.4/py3/hadoop-config/spark-defaults.conf:1]: any reason to change this blob?

File: spark/processing/3.4/py3/docker/py39/Dockerfile.cpu [hunk 3]
```diff
@@ -1,12 +1,14 @@
-FROM 137112412989.dkr.ecr.us-west-2.amazonaws.com/amazonlinux:2
+FROM public.ecr.aws/amazonlinux/amazonlinux:2
```
Comment (shenxiaoxu) [inline spark/processing/3.4/py3/docker/py39/Dockerfile.cpu:1]: ask to confirm that access public ecr will not create security issue on our side?

File: Makefile (hunk 1 above)
Comment (mahendruajay) [reply Makefile:37]: python vs python3 etc really depends upon the local python executable in your env.

File: spark/processing/3.4/py3/docker/py39/Dockerfile.cpu (hunk 3 above)
Comment (mahendruajay) [reply spark/processing/3.4/py3/docker/py39/Dockerfile.cpu:1]: we should have always used public.ecr.aws/amazonlinux/amazonlinux:2 as it has the latest publicly released AL2. EMR also uses the same: https://code.amazon.com/packages/AwsChicagoImageBuildResources/blobs/555b5468092d89cd12d38d577992b334481b3920/--/src/main/resources/release-build/spark3/vanilla/emr-6.15.0/Dockerfile?#L1

**Review by xiaoxshe — APPROVED** [review]
approved. I used a different account to review this PR.

**Pull request closed without merging**